test/state_module/on_state_entry_test.exs

```diff
@@ -14,11 +14,32 @@ defmodule StateServerTest.StateModule.OnStateEntryTest do
     def handle_call(action, _from, _state, _data), do: {:reply, :ok, action}
 
     @impl true
+    def handle_transition(:start, :tr_foo, pid) do
+      {:noreply, update: {"foo", pid}}
```

Comment:
so my original concern is with the above line: this `update` event does not change any data, since the 3rd arg matches `pid`, but it should be `{"foo", pid}`